src/Components/PaperEditor/ProgrammeScriptContainer/index.js

```diff
@@ -281,12 +281,64 @@ const ProgrammeScriptContainer = (props) => {
           };
           newElement.words.push(newWord);
         });
+
+        elementsClone.splice(insertElementIndex, 0, newElement);
+        const updatedElements = updateWordTimingsAfterInsert(elementsClone, insertElementIndex);
+        setElements(updatedElements);
+        setResetPreview(true);
+        handleSaveProgrammeScript(updatedElements);
+
+      } else {
+        const wordsArray = divideWordsSelectionsIntoParagraphs(result.words);
+
+        const elsToAdd = wordsArray.reduce((prevResult, array) => {
```

Comment:
there might be a better name choice for these variables to make it a little bit more descriptive :) `array` is not great as a choice, it's like naming something a `variable`, or `object` imo
```suggestion
        const paperEditElements = paragraphSelections.reduce((result, paragraphs) => {
```